gem5/HybridDatapath.cpp

```diff
@@ -516,12 +516,16 @@ bool HybridDatapath::handleDmaMemoryOp(ExecNode* node) {
 
   MemAccessStatus status = Invalid;
   unsigned node_id = node->get_node_id();
-  if (inflight_dma_nodes.find(node_id) == inflight_dma_nodes.end())
-    inflight_dma_nodes[node_id] = status;
-  else
+  if (inflight_dma_nodes.find(node_id) == inflight_dma_nodes.end()) {
+    if (inflight_dma_nodes.size() < maxInflightNodes)
```

Comment:
I don't think the statement "inflight_dma_nodes.size() < maxInflightNodes will never evaluate to true" is correct. Other DMA nodes could complete and erase their entries from inflight_dma_nodes. 

I'm also not sure what this PR actually changes in terms of behavior:
* Before, `handleDmaMemoryOp` would add the entry into the map and return false if the map's size > maxInflightNodes. Subsequent calls to this function will also return false since `status = Invalid` and the map's size is still too large.
* After, `handleDmaMemoryOp` won't add the entry at all; it will just return false; subsequent calls will do the same thing.

So at the end of the day, nothing has changed...am I missing something?

Reply:
Consider this example:

Let us say the queue size is 100 and it is currently full right now. The system receives, say, 101 new DMA requests which are set to `Invalid` and added to the queue (making the queue size 201), but not issued since `inflight_dma_nodes.size() > maxInflightNodes`. After some time, the initial 100 requests will all complete and leave the queue, but the condition `inflight_dma_nodes.size() < maxInflightNodes` will still be false, preventing any of the new 101 requests from being issued.

You are right in that the PR doesn't change anything behaviorally, but it prevents the problem I just described by only adding nodes to the queue when they can be processed.

Reply:
Ah, I see. Thanks for the explanation! Can you include a shortened version of this in a comment in the code for future readers?

Reply:
Done!